wordpress-mobile/WordPress-iOS
# Issues/update facebook sharing

This is an 11th hour patch to address Facebook's privacy changes that affect publicize sharing to Facebook Profiles.  As of August 1st, sharing to Facebook Profiles is no longer allowed.

This patch changes how Facebook sharing is handled so only sharing to Facebook Pages is supported.  Attempting to share to a profile will result in a notice.
![simulator screen shot - iphone 8 plus - 2018-07-28 at 15 50 48](https://user-images.githubusercontent.com/1435271/43360891-93cd4f10-9286-11e8-959a-f91d7d323aa4.png)

An existing connection to a Facebook profile will show up as broken.  On the detail page new messaging is shown.
![simulator screen shot - iphone 8 plus - 2018-07-28 at 16 47 33](https://user-images.githubusercontent.com/1435271/43360892-b6139df4-9286-11e8-8cee-fe395ddf9a07.png)

Clicking `Learn More` on either screen will open Safari to an FAQ.  The destination is different for each button and should match the behavior in calypso.

To test:
You'll want a Facebook account to test with. 

Scenario 1:
Attempt to set up a sharing connection to a Facebook profile.  You should see the prompt that the connection could not be made because their are no pages.
Confirm clicking the Learn More button opens the support page in Safari.

Scenario 2:
Create a Facebook Page.
Attempt to set up sharing to Facebook again, and this time confirm that your newly created page is shown as an option.  Go ahead and make the connection.

Scenario 3:
Edit `SharingDetailViewController.m` so `mustDisconnectFacebook` always returns true. 
Navigate to the Facebook connection details screen and confirm that instead of a "Reconnect" option, you see a "Learn More" option and the appropriate footer text.  
Confirm that the FAQ link opens in Safari.

FYI @loremattei 

@ctarda could I trouble you for a review?

cc @kwonye who is wrangling the Android version of this patch.
